src/pyhf/infer/intervals.py

```diff
@@ -0,0 +1,66 @@
+"""Interval estimation"""
+from . import hypotest
+from .. import get_backend
+import numpy as np
+
+
+def _interp(x, xp, fp):
+    tb, _ = get_backend()
+    return tb.astensor(np.interp(x, xp.tolist(), fp.tolist()))
+
+
+def upperlimit(data, model, scan, level=0.05, return_results=False):
```

Comment:
I think I would still go with `upper_limit` for the API, but this is something we should discuss or get feedback on.
```suggestion
def upper_limit(data, model, scan, level=0.05, return_results=False):
```